storage/db-tool/src/db_backup.rs

```diff
@@ -0,0 +1,209 @@
+        state_snapshot::backup::{StateSnapshotBackupController, StateSnapshotBackupOpt},
+        transaction::backup::{TransactionBackupController, TransactionBackupOpt},
+    },
+    coordinators::backup::{BackupCoordinator, BackupCoordinatorOpt},
+    metadata::{cache, cache::MetadataCacheOpt},
+    storage::StorageOpt,
+    utils::{
+        backup_service_client::{BackupServiceClient, BackupServiceClientOpt},
+        ConcurrentDownloadsOpt, GlobalBackupOpt,
+    },
+};
+use aptos_logger::{Level, Logger};
+use aptos_push_metrics::MetricsPusher;
+use clap::{Parser, Subcommand};
+use std::sync::Arc;
+
+#[derive(Subcommand)]
+pub enum Command {
+    #[clap(subcommand, about = "Manually run one shot commands.")]
+    OneShot(OneShotCommand),
+    #[clap(
+        subcommand,
+        about = "Long running process backing up the chain continuously."
+    )]
+    Coordinator(CoordinatorCommand),
+}
+
+#[derive(Parser)]
+pub enum OneShotCommand {
+    #[clap(
+        subcommand,
+        about = "Query the backup service builtin in the local node."
+    )]
+    Query(OneShotQueryType),
+    #[clap(about = "Do a one shot backup of either of the backup types.")]
+    Backup(OneShotBackupOpt),
+}
+
+#[derive(Parser)]
+pub enum OneShotQueryType {
+    #[clap(
+        about = "Queries the latest epoch, committed version and synced version of the local \
+        node, via the backup service within it."
+    )]
+    NodeState(OneShotQueryNodeStateOpt),
+    #[clap(
+        about = "Queries the latest epoch and versions of the existing backups in the storage."
+    )]
+    BackupStorageState(OneShotQueryBackupStorageStateOpt),
+}
+
+#[derive(Parser)]
+pub struct OneShotQueryNodeStateOpt {
+    #[clap(flatten)]
+    client: BackupServiceClientOpt,
+}
+
+#[derive(Parser)]
+pub struct OneShotQueryBackupStorageStateOpt {
+    #[clap(flatten)]
+    metadata_cache: MetadataCacheOpt,
+    #[clap(flatten)]
+    concurrent_downloads: ConcurrentDownloadsOpt,
+    #[clap(subcommand)]
+    storage: StorageOpt,
+}
+
+#[derive(Parser)]
+pub struct OneShotBackupOpt {
+    #[clap(flatten)]
+    global: GlobalBackupOpt,
+
+    #[clap(flatten)]
+    client: BackupServiceClientOpt,
+
+    #[clap(subcommand)]
+    backup_type: BackupType,
+}
+
+#[derive(Parser)]
+enum BackupType {
+    EpochEnding {
+        #[clap(flatten)]
+        opt: EpochEndingBackupOpt,
+        #[clap(subcommand)]
+        storage: StorageOpt,
+    },
+    StateSnapshot {
+        #[clap(flatten)]
+        opt: StateSnapshotBackupOpt,
+        #[clap(subcommand)]
+        storage: StorageOpt,
+    },
+    Transaction {
+        #[clap(flatten)]
+        opt: TransactionBackupOpt,
+        #[clap(subcommand)]
+        storage: StorageOpt,
+    },
+}
+
+#[derive(Parser)]
+pub enum CoordinatorCommand {
+    #[clap(
+        about = "Run the backup coordinator which backs up blockchain data continuously off \
+    a Aptos Node."
+    )]
+    Run(CoordinatorRunOpt),
```

Comment:
Looks like we don't need a second subcommand, let's roll it up.

Reply:
removed